server/controllers/question.js

```diff
@@ -0,0 +1,57 @@
+const Question = require("../models/question");
+const User = require("../models/user");
+var ObjectId = require("mongoose").Types.ObjectId;
+
+module.exports.getSingleQuestion = async (req, res, next) => {
+  const { difficulty } = req.body;
+  const { user } = req.user;
+  const userId = user._id;
+  console.log("user id:", userId);
+
+  try {
+    const user = await User.findById(userId);
```

Comment:
if you already have the user from the request, you don't need to find it again. Additionally you are currently redeclaring a constant here

Reply:
Here I have the user id (e.g. { _id: '608d9801716a735ea444d4d4' }) from the request not the whole user object, so I need to find the user from the db. Yes, I do not need to define a new variable. I can just use cons{ userId } = req.user._id .
I will apply this. Thank you.